patacrep/templates.py

```diff
@@ -84,11 +88,31 @@ def __init__(self, template, jinjaenv, encoding=None):
         self.jinjaenv.filters['escape_tex'] = _escape_tex
         self.jinjaenv.trim_blocks = True
         self.jinjaenv.lstrip_blocks = True
-        self.jinjaenv.filters["path2posix"] = files.path2posix
-        self.jinjaenv.filters["iter_datadirs"] = files.iter_datadirs
-        self.jinjaenv.filters["lang2babel"] = lang2babel
+        self._fill_filters()
         self.template = self.jinjaenv.get_template(template)
 
+    def _fill_filters(self):
+        """Define some jinja2 filters, if not set yet."""
+        for key, value in [
+                ("path2posix", files.path2posix),
+                ("iter_datadirs", files.iter_datadirs),
+                ("lang2babel", self.lang2babel),
```

Comment:
Le décorateur de filtre pourrait vérifier la présence d'une méthode `iter_errors` sur le résultat du filtre et logger ces erreurs le cas échéant.

Ainsi dans le cas de la langue, ça donnerait:
```py
class FallbackLanguage(Language):
    """Fallback language because the original code or babel lang is not supported."""

    def __init__(self, *, code, original_code=None, original_babel=None):
        self.code = code
        self.original_code = original_code
        self.original_babel = original_babel

    def iter_errors(self):
        yield SongUnknownLanguage(...)
        # en fonction des valeurs de original_code & original_babel
```